community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -250,9 +247,10 @@ public KernelTransactionImplementation( StatementOperationParts operations,
     /**
      * Reset this transaction to a vanilla state, turning it into a logically new transaction.
      */
-    public KernelTransactionImplementation initialize( long lastCommittedTx, long lastTimeStamp )
+    public KernelTransactionImplementation initialize( long lastCommittedTx, long lastTimeStamp,
+            StatementLocks statementLocks )
```

Comment:
i think we should not pass `statementLocks` here, we should pass all required params through constructor and build them ourselves